pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -157,6 +151,17 @@ func (fs *Decomposedfs) CreateStorageSpace(ctx context.Context, req *provider.Cr
 		return nil, err
```

Comment:
When you return here, the upper call to `SetMultiple()` has changed the nodes metadata.

Maybe it makes sense to let `SetMultiple` return a map of the previous values which can be used in a defer?

Reply:
i need to really read the attributes from the storage again. This is done in https://github.com/cs3org/reva/pull/2533/files#diff-d308a0558595437a537e77e29346472e66ab3bc31c213fcd2dd9b5f64c02aa91R154

Reply:
so the space has been successfully created already and the metadata has been set. If the AddGrant fails we should not delete the space again IMO.